SpatialGDK/Source/SpatialGDK/Private/SpatialView/EntityQuery.cpp

```diff
@@ -0,0 +1,114 @@
+﻿#include "SpatialView/EntityQuery.h"
+
+namespace SpatialGDK
+{
+
+EntityQuery::EntityQuery(const Worker_EntityQuery& Query)
+	: ResultType(static_cast<Worker_ResultType>(Query.result_type))
+{
+	Constraints.Reserve(GetNestedConstraintCount(Query.constraint));
+	Constraints.Add(Query.constraint);
+	StoreChildConstraints(Query.constraint, 0);
+	if (Query.result_type == WORKER_RESULT_TYPE_SNAPSHOT && Query.snapshot_result_type_component_ids)
+	{
+		SnapshotComponentIds.Reserve(Query.snapshot_result_type_component_id_count);
+  		SnapshotComponentIds.Append(Query.snapshot_result_type_component_ids, Query.snapshot_result_type_component_id_count);
+	}
+}
+
+Worker_EntityQuery EntityQuery::GetWorkerQuery() const
+{
+	return Worker_EntityQuery{
+		Constraints[0],
+		ResultType,
+		static_cast<uint32>(SnapshotComponentIds.Num()),
+		ResultType == WORKER_RESULT_TYPE_SNAPSHOT ? SnapshotComponentIds.GetData() : nullptr
+	};
+}
+
+int32 EntityQuery::GetNestedConstraintCount(const Worker_Constraint& Constraint)
+{
+	switch (static_cast<Worker_ConstraintType>(Constraint.constraint_type))
+	{
+	case WORKER_CONSTRAINT_TYPE_ENTITY_ID:
+	case WORKER_CONSTRAINT_TYPE_COMPONENT:
+	case WORKER_CONSTRAINT_TYPE_SPHERE:
+		return 1;
+	case WORKER_CONSTRAINT_TYPE_AND: {
+		auto& AndConstraint = Constraint.constraint.and_constraint;
+		size_t Sum = 1;
+		for (size_t i = 0; i < AndConstraint.constraint_count; ++i) {
```

Comment:
Nit: We use uint32 in StoreChildConstraints when looping to AndConstraint.constraint_count.

Reply:
Oh yeah. Forgot to convert that stuff.